pkg/webhooks/workload_webhook.go

```diff
@@ -182,6 +184,87 @@ func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateAdmissionChecks(obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i := range obj.Status.AdmissionChecks {
+		admissionChecksPath := basePath.Index(i)
+		ac := &obj.Status.AdmissionChecks[i]
+		allErrs = append(allErrs, validatePodSetUpdates(ac, obj, admissionChecksPath.Child("podSetUpdates"))...)
+	}
+	return allErrs
+}
+
+func validatePodSetUpdates(acs *kueue.AdmissionCheckState, obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	knowPodSets := make(sets.Set[string], len(obj.Spec.PodSets))
+	for i := range obj.Spec.PodSets {
+		name := obj.Spec.PodSets[i].Name
+		knowPodSets.Insert(name)
+	}
```

Comment:
I think  this can be replace with something like
```suggestion
	knowPodSets := sets.New(slices.Map(obj.Spec.PodSets,{// return name here})...)
```
and maybe do it only once it the caller function